src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/MtAmazonDynamoDbBySharedTable.java

```diff
@@ -506,7 +507,13 @@ public ScanResult scan(ScanRequest scanRequest) {
         if (!items.isEmpty()) {
             scanResult.setItems(items.stream().map(tableMapping.getItemMapper()::reverse).collect(toList()));
             if (scanResult.getLastEvaluatedKey() != null) {
-                scanResult.setLastEvaluatedKey(getKeyFromItem(Iterables.getLast(scanResult.getItems()), key));
+                Map<String, AttributeValue> lastItem = Iterables.getLast(scanResult.getItems());
+                Map<String, AttributeValue> lastEvaluatedKey = new HashMap<>();
+                lastEvaluatedKey.putAll(getKeyFromItem(lastItem, tableMapping.getVirtualTable().getPrimaryKey()));
+                if (scanRequest.getIndexName() != null) {
+                    lastEvaluatedKey.putAll(getKeyFromItem(lastItem, key));
+                }
+                scanResult.setLastEvaluatedKey(lastEvaluatedKey);
```

Comment:
If we're scanning a secondary index, the last evaluated key includes not only the secondary index PK fields, but also the table PK fields